cli/deployment/config.go

```diff
@@ -61,15 +110,19 @@ func (c Config) String() string {
 
 // The YAML function marshals a Config object to YAML. It sets all relative import paths to absolute paths and removes
 // all custom elements that the gcloud deployment manager is not aware of, including name, project, and description.
-func (c Config) YAML() ([]byte, error) {
+func (c Config) YAML(outputs map[string]map[string]interface{}) ([]byte, error) {
```

Comment:
Please update the doc comment to explain this new argument to the function. This would've made this review easier; I had to look through a lot of files to finally figure out that in fact this is an in/out parameter. Please also rename the argument to better reflect its purpose; as an in/out parameter, a name like "outputs" is misleading.